### PR TITLE
Abort merge if it fails

### PR DESCRIPTION
**Pull request by zsims**
This is on top of PR #15, and closes issue #14 by aborting the merge if it fails

### CODE REVIEW

**Comment by zsims** [comment]
@timn let me know what you think, and I'll merge :)

File: hooks/post-checkout [hunk 1]
```diff
@@ -34,7 +34,12 @@ merge() {
   GIT_COMMITTER_EMAIL="auto-merge@buildkite" \
   GIT_COMMITTER_NAME="github-merged-pr-buildkite-plugin"
   GIT_MERGE_AUTOEDIT="no" \
```
Comment (timn) [inline hooks/post-checkout]: This could be removed, as it is an older version of the --no-edit flag. But it is doesn't do any harm either.

Comment (zsims) [reply hooks/post-checkout]: Ahh cool, will opt to leave it per the "older scripts" comment in the docs.

Comment (zsims) [reply hooks/post-checkout]: On second thoughts will remove this... Not needed, and this option is confusing enough

**Pull request merged**